flexmeasures/data/models/planning/storage.py

```diff
@@ -250,11 +250,33 @@ def deserialize_flex_config(self):
         self.ensure_soc_min_max()
 
         # Now it's time to check if our flex configurations holds up to schemas
-        self.flex_model = StorageFlexModelSchema().load(self.flex_model)
+        self.flex_model = StorageFlexModelSchema(self.start).load(self.flex_model)
         self.flex_context = FlexContextSchema().load(self.flex_context)
 
+        # Extend schedule period in case a target exceeds its end
+        self.possibly_extend_end()
+
         return self.flex_model
 
+    def possibly_extend_end(self):
+        """Extend schedule period in case a target exceeds its end.
+
+        The schedule's duration is possibly limited by the server config setting 'FLEXMEASURES_MAX_PLANNING_HORIZON'.
+        """
+        soc_targets = self.flex_model.get("soc_targets")
+        if soc_targets:
+            max_target_datetime = max(
+                [soc_target["datetime"] for soc_target in soc_targets]
+            )
+            if max_target_datetime > self.end:
+                max_server_horizon = current_app.config.get(
+                    "FLEXMEASURES_MAX_PLANNING_HORIZON"
+                )
+                if max_server_horizon:
+                    self.end = min(max_target_datetime, self.start + max_server_horizon)
+                else:
```

Comment:
This could perhaps go, since we already throw a ValidationError in the StorageFlexModelSchema, but I thought to keep it as an extra safeguard for now.

I imagine that later `deserialize_flex_config` will become a single schema for the whole scheduler, in which `possibly_extend_end()` would become a class method with a `@post_load` decorator.

Reply:
Good to have a safeguard, maybe with a comment.